core/psa_crypto.c

```diff
@@ -7612,6 +7612,12 @@ static psa_status_t psa_key_derivation_input_internal(
     psa_status_t status;
     psa_algorithm_t kdf_alg = psa_key_derivation_get_kdf_alg(operation);
 
+    if (kdf_alg == 0) {
```

Comment:
```suggestion
    if (kdf_alg == PSA_ALG_NONE) {
```
3 occurences.

I think the intent of @gilles-peskine-arm's [comment ](https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/186#discussion_r1975742965
)was to keep it consistent between codebase and tests? 

It woudl be good for users revewing the code to understand that this exit's purpose is.